### PR TITLE
feat: Make tags editable

### PR DESCRIPTION
**Pull request by vikasgurjar**
Long awaited Edit tag option 
fixes https://github.com/react-tags/react-tags/issues/375

### CODE REVIEW

**Comment by vikasgurjar** [comment]
I have been using this library for a while now. Faced a requirement to have editable tags. So thought I will contribute to it

File: src/components/ReactTags.js [hunk 1]
```diff
@@ -367,13 +381,17 @@ class ReactTags extends Component {
     }
 
     // call method to add
-    this.props.handleAddition(tag);
+    if (currentEditIndex !== null && this.props.handleUpdateTag)
```
Comment (ad1992) [inline src/components/ReactTags.js:384]:
```suggestion
    if (!currentEditIndex && handleUpdateTag)
```
`handleUpdateTag` can be destructed on line 361

File: src/components/ReactTags.js [hunk 2]
```diff
@@ -367,13 +381,17 @@ class ReactTags extends Component {
...
+      this.props.handleUpdateTag(currentEditIndex, tag);
+    else
+      this.props.handleAddition(tag);
 
     // reset the state
     this.setState({
       query: '',
       selectionMode: false,
       selectedIndex: -1,
+      currentEditIndex: null,
```
Comment (ad1992) [inline src/components/ReactTags.js:394]: lets keep this -1 instead of `null` so its always numeric

File: src/components/ReactTags.js [hunk 3]
```diff
@@ -403,17 +421,37 @@ class ReactTags extends Component {
...
-      return (
+      if (currentEditIndex === index) {
+        return (
+          <div className={classNames.editTagInput}>
+            <input
+              ref={(input) => {
+                this.tagInput = input;
+              }}
+              key={index}
+              d-index={index}
```
Comment (ad1992) [inline src/components/ReactTags.js:443]: what is this used for ? are you trying to use `data-attribute` ?

**Review by ad1992 — CHANGES_REQUESTED** [review]
@vikasgurjar thanks a lot for adding this feature, it's is indeed one of the most awaited features.
can you also add specs for the changes?
I pushed a small fix so editable tags appear inline.

**Comment by ad1992** [comment]
@vikasgurjar any update on this ?

File: src/components/ReactTags.js (hunk 1 above)
Comment (vikasgurjar) [reply src/components/ReactTags.js:384]: Wouldn't this cause issue when we try to edit tag at index 0?

**Comment by vikasgurjar** [comment]
> @vikasgurjar thanks a lot for adding this feature, it's is indeed one of the most awaited features.
> can you also add specs for the changes?
> I pushed a small fix so editable tags appear inline.

What do you mean by "add specs". I'm not aware of that

File: src/components/ReactTags.js (hunk 1 above)
Comment (ad1992) [reply src/components/ReactTags.js:384]: good point, lets make it `currentEditIndex != -1` instead

**Comment by ad1992** [comment]
> > @vikasgurjar thanks a lot for adding this feature, it's is indeed one of the most awaited features.
> > can you also add specs for the changes?
> > I pushed a small fix so editable tags appear inline.
> 
> What do you mean by "add specs". I'm not aware of that

 I meant unit tests. We will need to add unit tests for the edit tag feature. No worries, I will try to add it once I get some time

File: src/components/ReactTags.js [hunk 4]
```diff
@@ -207,9 +212,16 @@ class ReactTags extends Component {
     });
   }
 
-  handleTagClick(i, e) {
-    if (this.props.handleTagClick) {
-      this.props.handleTagClick(i, e);
+  handleTagClick(i, tag, e) {
+    const { editable, handleTagClick, labelField } = this.props;
+    // const currentTagText=tags[i].text;
```
Comment (ad1992) [inline src/components/ReactTags.js:217]:
```suggestion
```

**Comment by ad1992** [comment]
@vikasgurjar thanks for the contribution, merging 🎉

**Pull request merged**